contracts/QuestNFT.sol

```diff
@@ -8,32 +8,56 @@ import "@openzeppelin/contracts/security/Pausable.sol";
 import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
 import 'base64-sol/base64.sol';
 
-contract QuestNFT is ERC721, Ownable, Pausable, MerkleProof {
-    uint256 publicMintPrice;
+contract QuestNFT is ERC721, Ownable, Pausable {
+    uint256 public publicMintPrice;
     uint256 internal _nextId;
+    uint256 public registerQuestCost;
+
+    // ============ QUEST REGISTRY ============
 
     mapping(uint256 => uint256) xpByTokenId;
     mapping(uint256 => mapping(uint256 => bool)) questCompletedByTokenId;
 
-    // Quest Registry
     struct Quest {
         uint256 questId;
         address questCreator;
         uint256 prerequisiteQuest;
         uint256 questRewardXP;
         address[] questPlayers;
         bytes4[] questGoals;
+        // needs to be an array
+        QuestParams questParams;
+        // reward logic
     }
 
-    function mint(address to) public payable publicMintPaid {
-        _mint(to, _nextId++);
+    enum QuestParams {
+
+    }
+
+    Quest[] public quests;
```

Comment:
does quest ID always match index in the array?

Reply:
yes